tests/models/test_data.py

```diff
@@ -53,37 +75,59 @@ def __init__(self):
                 self.data_files = []
                 self.normalizing_dict = norm_dict if normalize else None
                 self.to_tensor = None
+                self.experiment = experiment
 
         base_iterator = _BaseIter(MockLoader())
 
-        arrays = base_iterator.ds_folder_to_np(tmp_path, return_latlons=True,
+        arrays = base_iterator.ds_folder_to_np(data_dir, return_latlons=True,
                                                to_tensor=to_tensor)
 
-        x_np, y_np, latlons = arrays.x.historical, arrays.y, arrays.latlons
+        x_train_data, y_np, latlons = (
+            arrays.x, arrays.y, arrays.latlons
+        )
+        assert isinstance(x_train_data, TrainData)
+        if not to_tensor:
+            assert isinstance(y_np, np.ndarray)
 
         if to_tensor:
-            assert (type(x_np) == torch.Tensor) and (type(y_np) == torch.Tensor)
+            assert (
+                type(x_train_data.historical) == torch.Tensor
+            ) and (type(y_np) == torch.Tensor)
         else:
-            assert (type(x_np) == np.ndarray) and (type(y_np) == np.ndarray)
-
-        assert x_np.shape[0] == y_np.shape[0] == latlons.shape[0], \
-            f'x, y and latlon data have a different number of instances! ' \
-            f'x: {x_np.shape[0]}, y: {y_np.shape[0]}, latlons: {latlons.shape[0]}'
+            assert (
+                type(x_train_data.historical) == np.ndarray
+            ) and (type(y_np) == np.ndarray)
+
+        if (not normalize) and (experiment == 'nowcast') and (not to_tensor):
+            assert (
+                x_train_data.historical.shape[0] == x_train_data.current.shape[0]
+            ), "The 0th dimension (latlons) should be equal in the " \
+                f"historical ({x_train_data.historical.shape[0]}) and " \
+                f"current ({x_train_data.current.shape[0]}) arrays."
+
+            expected = (
+                x.precip
+                .sel(time=y.time)
+                .stack(dims=['lat', 'lon'])
+                .values.T
+            )
+            got = x_train_data.current
+            assert expected.shape == got.shape, "should have stacked latlon" \
+                " vars as the first dimension in the current array."
+
+            assert all(expected == got), "" \
```

Comment:
@gabrieltseng this is the major issue. I am sure i am really close but something somewhere is going wrong. Can you just check my test logic and whether i am selecting the right array to compare to?

I am hoping to get `precip` and the final timestep (`time=-1`) and the values stacked (so a `(25,1)` array and that is what should be the `current` vector. but for some reason the value returned from the DataLoader isnot the same?

Any help very much appreciated sorry for the pain!